gnoblog-cli/cmd/post.go

```diff
@@ -0,0 +1,261 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/gnolang/gno/gno.land/pkg/gnoclient"
+	"github.com/gnolang/gno/gnovm/pkg/gnoenv"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/peterbourgon/ff/v3/ffcli"
+	"os"
+	"strings"
+	"time"
+)
+
+type cliCfg struct {
+	Publish       bool
+	Edit          bool
+	GasWanted     int64
+	GasFee        string
+	ChainId       string
+	BlogRealmPath string
+
+	KeyName               string
+	GnoHome               string
+	Remote                string
+	Quiet                 bool
+	InsecurePasswordStdIn bool
+}
+
+func newPostCommand() *ffcli.Command {
+	var (
+		fs  = flag.NewFlagSet("post", flag.ExitOnError)
+		cfg = &cliCfg{}
+	)
+
+	// Register flagset
+	cfg.registerFlags(fs)
+
+	// Make Post command
+	return &ffcli.Command{
+		Name:       "post",
+		ShortUsage: "post <FILE OR FILES_DIR> [flags]",
+		LongHelp:   `Post one or more files. Passing in a file will post that single file, while passing in a directory will search for all .md files and batch post them.`,
+		FlagSet:    fs,
+		Exec:       cfg.execPost,
+	}
+}
+
+// Register flags
+func (cfg *cliCfg) registerFlags(fs *flag.FlagSet) {
+	fs.StringVar(&cfg.KeyName,
+		"key",
+		"",
+		"name of keypair to use for deployment",
+	)
+	fs.BoolVar(&cfg.Publish,
+		"publish",
+		false,
+		"publish blogpost",
+	)
+	fs.BoolVar(&cfg.Edit,
+		"edit",
+		false,
+		"edit mode",
+	)
+	fs.Int64Var(&cfg.GasWanted,
+		"gas-wanted",
+		5000000,
+		"gas requested for tx",
+	)
+	fs.StringVar(&cfg.GasFee,
+		"gas-fee",
+		"1000000ugnot",
+		"gas payment fee",
+	)
+	fs.StringVar(&cfg.ChainId,
+		"chainid",
+		"dev",
+		"chain ID",
+	)
+	fs.StringVar(&cfg.BlogRealmPath,
+		"pkgpath",
+		"gno.land/r/gnoland/blog",
+		"blog realm path",
+	)
+	fs.StringVar(&cfg.GnoHome,
+		"home",
+		gnoenv.HomeDir(),
+		"home directory",
+	)
+	fs.StringVar(&cfg.Remote,
+		"remote",
+		"localhost:26657",
+		"remote node URL",
+	)
+	fs.BoolVar(&cfg.InsecurePasswordStdIn,
+		"insecure-password-stdin",
+		false,
+		"WARNING! take password from stdin",
+	)
+}
+
+func (cfg *cliCfg) execPost(_ context.Context, args []string) error {
+	if len(args) > 1 {
+		fmt.Println(args)
+
+		return errInvalidNumberOfArgs
+	}
+
+	fileInfo, err := os.Stat(args[0])
+	if err != nil {
+		return errInvalidPath
+	}
+
+	// Init IO for password input
+	io := commands.NewDefaultIO()
```

Comment:
It would be better if you run this from the main and use our commands library instead of ffcli. This way, helpers can focus more on their goal, and the main function can focus on initializing everything.

Reply:
I moved the IO to main: [3399548](https://github.com/gnolang/blog/pull/63/commits/3399548962677fb645c74e46745cc8f5c4a489f2)

I'll see to use our commands lib.

Reply:
Using our commands lib: [7f1aa96](https://github.com/gnolang/blog/pull/63/commits/7f1aa965d5a98cf555ce05be56f44ecf132e73a9)